fabric-climbable-v1/src/main/java/net/fabricmc/fabric/mixin/entity/MixinLivingEntity.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.entity;
+
+import net.fabricmc.fabric.api.block.Climbable;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(LivingEntity.class)
+public abstract class MixinLivingEntity {
+
+    @Inject(method = "isClimbing", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/BlockState;getBlock()Lnet/minecraft/block/Block;", shift = At.Shift.AFTER), locals = LocalCapture.CAPTURE_FAILHARD, cancellable = true)
+    public void isClimbing(CallbackInfoReturnable<Boolean> cir, final BlockState state) {
+
+        final Climbable climbable = (Climbable) state.getBlock();
+        final LivingEntity thisLivingEntity = (LivingEntity) (Object) this;
+
+        Climbable.ClimbBehavior behavior = climbable.canClimb(thisLivingEntity, state, thisLivingEntity.getBlockPos());
+        if (behavior != Climbable.ClimbBehavior.Vanilla) {
+
+			if (behavior == Climbable.ClimbBehavior.True) {
+				cir.setReturnValue(true);
+			}
+			else {
+				cir.setReturnValue(false);
+			}
+			cir.cancel();
```

Comment:
Which causes the engine-special casing to get ignored; the whole point of only setting the return value only when it's true or false is so that if `canClimb()` returns `ClimbableBehavior.Vanilla`, the game ignores that result and continues on with its own checks.

Reply:
Oh, that's what you were talking about... No, you misunderstood what I meant. You do not remove the `behavior != VANILLA` (or `behavior != DEFAULT` for a TriState). Notice I said to remove exactly 7 lines.